koku/koku/middleware.py

```diff
@@ -64,7 +64,13 @@
 
 def is_no_auth(request):
     """Check condition for needing to authenticate the user."""
-    no_auth_list = ["/status", "openapi.json", "/metrics"]
+    no_auth_list = [
+        "/status",
+        "openapi.json",
+        "/metrics",
+        "/recommendations/openshift",
+        "/recommendations/openshift/{recommendation-id}",
```

Comment:
```suggestion
        "/recommendations/openshift/openapi.json"
```
We need to add only openapi.json API under no_auth_list